### PR TITLE
feat: add partner counts to partner follow button

### PR DESCRIPTION
**Pull request by egdbear**
This PR resolves [DIA-131] 

### PR Checklist

- [X] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [X] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

Screenshot 2023-09-19 at 22 16 06

OOPS
image

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add partner counts to partner follow button - egdbear

#### iOS user-facing changes

-

#### Android user-facing changes
-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-131]: https://artsyproduct.atlassian.net/browse/DIA-131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b766883e4510d1d55368b18c12caeffc6f9a5e0f
</p>

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  const { renderWithRelay } = setupTestWrapper<PartnerFollowButtonTestQuery>({
+    Component: ({ partner }) => <PartnerFollowButtonFragmentContainer partner={partner!} />,
+    query: graphql`
+      query PartnerFollowButtonTestQuery @relay_test_operation {
+        partner(id: "white-cube") {
+          ...PartnerFollowButton_partner
+        }
+      }
+    `,
+  })
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx:16]: 🔥

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          name: "White Cube",
+          isFollowed: true,
+          counts: {
+            follows: 670,
+          },
+        },
+      }),
+    })
+
+    expect(queryByText(/Following 670/)).toBeTruthy()
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx:31]: **suggestion(non-blocking):** 
```suggestion
    expect(queryByText(/Following 670/)).toBeOnTheScreen()
```

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          name: "White Cube",
+          isFollowed: true,
+          counts: {
+            follows: 499,
+          },
+        },
+      }),
+    })
+
+    expect(queryByText(/Following/)).toBeTruthy()
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx:47]: suggestion(non-blocking):
```suggestion
    expect(queryByText(/Following/)).toBeOnTheScreen()
```

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      Partner: () => ({
+        profile: {
+          name: "White Cube",
+          isFollowed: false,
+          counts: null,
+        },
+      }),
+    })
+
+    expect(queryByText(/Follow/)).toBeTruthy()
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerFollowButton.tests.tsx:61]: **suggestion(non-blocking):**
```suggestion
    expect(queryByText(/Follow/)).toBeOnTheScreen()
```

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tsx [hunk 5]
```diff
@@ -84,13 +90,16 @@ export class PartnerFollowButton extends React.Component<Props, State> {
 
   render() {
     const { partner } = this.props
-
+    const hasFollows = partner.profile?.counts?.follows >= 500
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerFollowButton.tsx:93]: 🤯 so we're not showing that for artists with less than 500 follows 🤔

**Review by gkartalis — DISMISSED** [review]
Super

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tsx (hunk 5 above)
Comment (egdbear) [reply src/app/Scenes/Partner/Components/PartnerFollowButton.tsx:93]: yes it should have minimum of 500 followers [see here](https://artsyproduct.atlassian.net/browse/DIA-131)

**Review by gkartalis — APPROVED** [review]
🛳️

File: src/app/Scenes/Partner/Components/PartnerFollowButton.tsx (hunk 5 above)
Comment (MounirDhahri) [inline src/app/Scenes/Partner/Components/PartnerFollowButton.tsx:93]: **nitpick:** I would suggest going with `showFollows` instead of `hasFollows` here

**Review by MounirDhahri — COMMENTED** [review]
Nice thanks for adding this

**Pull request merged**